sudoku_board.h

```diff
@@ -0,0 +1,40 @@
+#pragma
+#include <vector>
+#include <set>
+#include <iostream>
+#include <memory>
+#include <algorithm>
+
+
+enum STATUS_OPTION {WIN, LOSE, PROG};
+
+class SudokuBoard {
+private:
+    int m_size;
+    std::vector<std::vector<int>> m_board_values;
+    std::vector<std::vector<std::set<int>>> m_possible_options;
+
+    std::set<int> get_values_in_cube(int r, int c);
+    std::set<int> get_values_in_row(int r, int c);
+    std::set<int> get_values_in_col(int r, int c);
+    std::set<int> get_forbidden_values(int r, int c);
+    void empty_options();
+public:
+    const int EMPTY_CELL_SIGN = 0;
+    const int CUBE_SIZE = 3;
+    const std::set<int> POSSIBLE_CELL_VALUES = {1, 2, 3, 4, 5, 6, 7, 8, 9};     
+
+    SudokuBoard(int size);
+    SudokuBoard(int size, std::vector<std::vector<int>> board_values);
+    int get_size();
+    std::vector<std::vector<int>> get_board_values();
+    std::vector<std::vector<std::set<int>>> get_possible_options();
+    std::set<int> get_cell_options(int r, int c);
+    STATUS_OPTION get_status();
```

Comment:
```suggestion
    STATUS_OPTION get_status() const;
```
your function doesn't change the board state, please make it const. This usually doesn't have any effect, but does 2 things: 1) Makes sure the board doesn't change, this can resolve bugs. 2) The compiler can futher optimize the code.
Put in your mind - Everything by default should be const, make stuff non-const only when you want to change them.